[Improvement-17248][UI] Optimize file details display area to reduce blank space

zhan7236 opened the pull request:
### What is the purpose of the pull request

Fix #17248

This PR optimizes the file details display area in the Resource Center to reduce blank space and ensure buttons are always visible.

### Brief change log

- Use viewport-based height calculation (`calc(100vh - Npx)`) for Monaco Editor to dynamically adapt to different screen sizes
- Remove complex flexbox layout that caused infinite scrolling issues  
- Reduce Monaco Editor `minHeight` from 200px to 150px for better small screen support
- Ensure "Save" and "Return" buttons remain visible without scrolling on normal screen sizes

### Verify this pull request

This pull request was tested manually by:
1. Opening the file create page (`/resource/file/create`)
2. Opening the file edit/detail page
3. Verifying buttons are visible at the bottom without scrolling
4. Testing with different browser window sizes to ensure responsive behavior
5. Confirming that only on very small screens does scrolling become necessary (expected behavior)

### Screenshots

<img width="1920" height="919" alt="12c93d5090fbb92bda2f9e289afd63b3" src="https://github.com/user-attachments/assets/7db6dc18-2978-4ee6-84e4-c90c79fe83ef" />
<img width="877" height="708" alt="37263aae9d2d967a0c8879abbea0ca14" src="https://github.com/user-attachments/assets/3539b081-e1bc-43f2-88ae-cc4c9fd26902" />


### Related Issues

Closes #17248

zhan7236 committed:
[Improvement-17248][UI] Optimize file details display area to reduce blank space

- Add height prop to MonacoEditor component with default value '300px'
- Use dynamic height calculation for file edit page: calc(100vh - 320px)
- Use dynamic height calculation for file create page: calc(100vh - 420px)
- Ensures MonacoEditor fills the available screen space
- Improves user viewing experience by reducing large blank space

This resolves the issue where the file details display area had
excessive blank space at the bottom.

Closes #17248

zhan7236 committed:
refactor: address Copilot AI code review feedback

- Extract hard-coded magic numbers (320px, 420px) to named constants
- Add minHeight constraint (200px) to MonacoEditor for better UX on small screens
- Document offset values with detailed comments explaining what UI elements contribute to each offset

Fixes #17248

SbloodyS committed:
Merge branch 'dev' into Improvement-17248

SbloodyS reviewed (commented):
![OKIkGovMxXf6H2C.png](https://cdn.sa.net/2025/11/26/OKIkGovMxXf6H2C.png)

The following two buttons are beyond the current page, and need to be slid down to be fully displayed. It is best that just one screen does not need to be slid down.

BTW, Please run `pnpm run lint` to format the code. @zhan7236

zhan7236 committed:
style: fix trailing whitespace in create/index.tsx

zhan7236 committed:
refactor: use Flexbox layout to optimize file editor display area

- Replace hardcoded pixel offsets (calc(100vh - 320px)) with Flexbox layout
- Add .file-edit-content container with flex column layout
- Add .editor-wrapper with flex: 1 to auto-fill remaining space
- Add .button-area with flex-shrink: 0 to keep buttons always visible
- Remove constants.ts as fixed pixel offsets are no longer needed
- Add minHeight: 200px to ensure editor has minimum usable height

This approach is more robust than fixed pixel calculations:
- Automatically adapts to different screen sizes
- Buttons are always visible without scrolling
- No need to manually calculate header/footer heights

Fixes #17248

zhan7236 commented:
Sorry, I need to make a further fix.

SbloodyS commented:
> Sorry, I need to make a further fix.

It's ok. You can take your time.

zhan7236 committed:
fix: add overflow hidden to prevent editor from infinite expansion

- Add overflow: hidden to .file-edit-content container
- Add overflow: hidden to .form-content
- Add overflow: hidden and max-height to .editor-wrapper
- This ensures the editor stays within bounds and buttons remain visible

zhan7236 commented:
I have re-fixed the code. Could you please take some time to review it when you're available? Thank you!

SbloodyS committed:
Merge branch 'dev' into Improvement-17248

SbloodyS commented:
> I have re-fixed the code. Could you please take some time to review it when you're available? Thank you!

The slid issue still exists. You can use the mouse wheel or the keyboard to slide up and down. Part of the lowered button is blocked by sliding.

![ZjnOg6S1P3WyYis.png](https://cdn.sa.net/2025/11/27/ZjnOg6S1P3WyYis.png)

zhan7236 committed:
fix: use viewport-based height calculation for Monaco Editor to ensure buttons are always visible

- Use calc(100vh - Npx) to dynamically calculate editor height based on viewport
- Remove complex flexbox layout that caused infinite scrolling issues
- Reduce Monaco Editor minHeight from 200px to 150px for better small screen support
- Ensure Save and Return buttons remain visible without scrolling on normal screen sizes

SbloodyS committed:
Merge branch 'dev' into Improvement-17248

zhan7236 committed:
Merge branch 'dev' into Improvement-17248

SbloodyS committed:
Merge branch 'dev' into Improvement-17248

zhan7236 committed:
fix: run pnpm lint to fix code format

SbloodyS committed:
Merge branch 'dev' into Improvement-17248

SbloodyS reviewed (approved):
LGTM
